packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2817,6 +2817,12 @@ export class ContainerRuntime
 				forcedFullTree,
 			} as const;
 
+			try {
+				this.summarizerNode.validateSummary();
+			} catch (error) {
+				return { stage: "generate", ...generateSummaryData, error };
+			}
```

Comment:
Another important reason to resume the inbound queue is for scenarios where local data stores are created. If their handles are stored in a DDS, they will move to `Attaching` state. `summarize` and `getGCData` cannot work with data stores in `Attaching` state because we cannot reason about these data stores so we fail summarize.
In order for the next attempt to be successful, these data store have to move to `Attached` state which can only happen if their attach op round trips.

Reply:
> return extra information in the result as part of the "generate" stage

This makes sense, and at the same time I'd argue the "generate" stage is the one that failed, so putting stage = "generate" here could be a little misleading when aggregating telemetry. Because I think that typically means that "generate" is the last step that succeeded right?

Maybe we can add another flag to `IGeneratedSummaryStats` that indicates that the Summary was successfully generated but was invalid.

Reply:
You are right - now that the validate stage runs before upload, techically the generate stage failed. We can return "base" here. Although I am not sure I like base because it doesn't mean anything.

Reply:
Returning "base" here now.